core/src/test/java/io/grpc/internal/ManagedChannelImplTest.java

```diff
@@ -587,10 +595,50 @@ public void callOptionsExecutor() {
   @Test
   public void nameResolutionFailed() {
     Status error = Status.UNAVAILABLE.withCause(new Throwable("fake name resolution error"));
+    FakeNameResolverFactory nameResolverFactory = new FakeNameResolverFactory(error);
+    // Name resolution is started as soon as channel is created.
+    createChannel(nameResolverFactory, NO_INTERCEPTOR);
+    FakeNameResolverFactory.FakeNameResolver resolver = nameResolverFactory.resolvers.get(0);
+    verify(mockLoadBalancer).handleNameResolutionError(same(error));
 
+    timer.forwardNanos(RECONNECT_BACKOFF_INTERVAL_NANOS);
+
+    assertEquals(1, resolver.refreshCalled);
+    verify(mockLoadBalancer, times(2)).handleNameResolutionError(same(error));
+
+    // Allow the next refresh attempt to succeed
+    resolver.error = null;
+
+    // For the second attempt, the backoff should occur at RECONNECT_BACKOFF_INTERVAL_NANOS * 2
+    timer.forwardNanos(RECONNECT_BACKOFF_INTERVAL_NANOS);
```

Comment:
Ditto. The forward time here should be RECONNECT_BACKOFF_INTERVAL_NANOS * 2 - 1, and the next forward should be 1.

Reply:
Done